jenkins/Jenkinsfile-blossom.premerge

```diff
@@ -268,101 +257,27 @@ pipeline {
-                            cloud 'sc-ipp-blossom-prod'
-                            yaml "${IMAGE_DB}"
-                        }
-                    }
-                    environment {
-                        DB_RUNTIME = '11.3'
-                        DATABRICKS_RUNTIME = DbUtils.getRuntime("$DB_RUNTIME")
-                        BASE_SPARK_VERSION = DbUtils.getSparkVer("$DB_RUNTIME")
-                        BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS = DbUtils.getInstallVer("$DB_RUNTIME")
-                        INIT_SCRIPTS = DbUtils.getInitScripts("$DB_RUNTIME")
-                    }
-                    steps {
-                        script {
-                            timeout(time: 6, unit: 'HOURS') {
-                                unstash "source_tree"
-                                databricksBuild()
-                            }
-                        }
-                    }
-                } // end of DB runtime 11.3
-
-                stage('DB runtime 12.2') {
-                    when {
-                        beforeAgent true
-                        anyOf {
-                            expression { db_build }
-                        }
-                    }
-
-                    agent {
-                        kubernetes {
-                            label "premerge-ci-db-12.2-${BUILD_NUMBER}"
-                            cloud 'sc-ipp-blossom-prod'
-                            yaml "${IMAGE_DB}"
-                        }
-                    }
-                    environment {
-                        DB_RUNTIME = '12.2'
-                        DATABRICKS_RUNTIME = DbUtils.getRuntime("$DB_RUNTIME")
-                        BASE_SPARK_VERSION = DbUtils.getSparkVer("$DB_RUNTIME")
-                        BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS = DbUtils.getInstallVer("$DB_RUNTIME")
-                        INIT_SCRIPTS = DbUtils.getInitScripts("$DB_RUNTIME")
-                    }
+                stage('Databricks') {
                     steps {
                         script {
-                            timeout(time: 6, unit: 'HOURS') {
-                                unstash "source_tree"
-                                databricksBuild()
+                            try {
+                                build(job: 'tim-rapids-databricks_premerge-github1',
+                                    propagate: true, wait: true,
+                                    parameters: [
+                                            string(name: 'REF', value: params.REF),
+                                            string(name: 'GITHUB_DATA', value: params.GITHUB_DATA)
+                                    ])
+                            } catch (e){
+                                // 'JOB_NAME #ID completed with status FAILURE xxx' --> 'JOB_NAME/ID'
+                                def dbJobId = e.getCauses()[0].getShortDescription().split('completed with status FAILURE')[0].replace(' #', '/').trim()
```

Comment:
nit: can we try achieve this w/o exception catch ops? probably like

set `propagate: false` and use returned downstream_job (jenkins data model) to control the workflow
```
def DBJob = build(...)
if ( DBJob.result == 'SUCCESS' ) { // https://javadoc.jenkins.io/hudson/model/Result.html#field.summary
// ...
} else {
// do some ops use fields at https://javadoc.jenkins.io/hudson/model/Run.html?is-external=true#field.summary
}
```

Reply:
Good point, let me update it.